### PR TITLE
lodash: _.get should work with NumericDictionaries

### PR DESCRIPTION
**Pull request by aj-r**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: N/A
- [ ] Increase the version number in the header if appropriate. N/A
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by mhegazy** [comment]
@bczengel, @chrootsu, @stepancar, @ericanderson, @ailrun, @e-cloud, @thorn0, @jtmth, @rzymek @unional @OliverJAsh @sandersn @polizz @falsandtru @andnp @minestarks @thorn0 @jtmthf @JKillian @bcherny @yuit @tamird @e-cloud @vaskevich @jsheely @Ailrun @evestraw @ericanderson @danvk  can you please review

File: types/lodash/lodash-tests.ts [hunk 1]
```diff
@@ -11266,6 +11266,22 @@ namespace TestGet {
     _.get([], Symbol.iterator);
     _.get([], [Symbol.iterator]);
 
+    _.get('abc', 4); // $ExpectType string
+    _('abc').get(4); // $ExpectType string
+    _.chain('abc').get(4); // $ExpectType LoDashExplicitWrapper<string>
+
+    _.get({ a: { b: true } }, "a"); // $ExpectType { b: boolean; }
```
Comment (sandersn) [inline types/lodash/lodash-tests.ts]: How are these following tests related to the addition of NumericDictionary overloads? I didn't think object literals had numeric index signatures, and `"a"` isn't of type number, either.

File: types/lodash/lodash-tests.ts [hunk 2]
```diff
@@ -11266,6 +11266,22 @@ namespace TestGet {
     _.get([], Symbol.iterator);
     _.get([], [Symbol.iterator]);
 
+    _.get('abc', 4); // $ExpectType string
+    _('abc').get(4); // $ExpectType string
+    _.chain('abc').get(4); // $ExpectType LoDashExplicitWrapper<string>
```
Comment (sandersn) [inline types/lodash/lodash-tests.ts]: Seems a little strange to have the test get an out-of-bounds index. Why not 1 or 2?

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good; just a couple of questions.

File: types/lodash/lodash-tests.ts (hunk 1 above)
Comment (aj-r) [reply types/lodash/lodash-tests.ts]: Yeah, they're not related to `NumericDictionary`. I added them because it seemed that we were lacking tests for the `object` overloads.

**Review by sandersn — APPROVED** [review]
